blockchain/tx_pool.go

```diff
@@ -533,9 +533,14 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 		pool.gasPrice = misc.NextMagmaBlockBaseFee(newHead, pset.ToKip71Config())
 	}
 
+	pending, err := pool.Pending()
```

Comment:
We cannot call `Pending` here since `reset` already holds a lock. Maybe we can introduce internal `pendingLocked` that can be used here and `Pending` also.

Reply:
I checked that
- `pool.mu` is never acquired in the `reset()` function (L502-503) (perhaps inside individual functions like addTxsLocked -> add)
- `pool.txMu` is locked only within selective paragraphs.